user_guide_src/source/libraries/uploaded_files.rst

```diff
@@ -69,9 +69,7 @@ In this case, the returned array of files would be more like::
 
 If you just need to access a single file, you can use ``getFile()`` to retrieve the file instance directly::
 
-	$file = $this->request->getFile('avatar');
-
-.. note:: This currently only works with simple file names and not with array syntax naming.
+	$file = $this->request->getFile('my-form.details.avatar.0');
```

Comment:
A couple of things: 

1. We should still show that you can access the simple case `$this->request->getFile('avatar');`. 
2. If there are multiple files with the same name, then your example applies, but what if there's only one file with a deep name? Basically - we should show all three methods in the docs. 

```
// Get a single file
$file = $this->request->getFile('avatar');
// Get a single file within a more complex path
$file = $this->request->getFile('my-form.details.avatar');
// Get the first of several files uploaded under the same name
$file = $this->request->getFile('my-form.details.avatar.0');
```

Looking at the code, I believe those cases will all pass currently, and we need them to. :)